Lovecalculator gives error upon using command

Trying to get the lovecalculator to work but every time i run the command i get the same issue
I installed BeautifulSoup thinking it would fix it but it didn't. Is the cog just outdated perhaps? Would love to see it work but from my tiny amount of coding knowledge i couldn't find any reason why it wouldn't work

` [2020-06-16 15:04:09] [ERROR] red: Exception in command 'lovecalculator'
Traceback (most recent call last):
  File "c:\programs\redenv\lib\site-packages\discord\ext\commands\core.py", line 83, in wrapped
    ret = await coro(*args, **kwargs)
  File "C:\Users\Kaval\AppData\Local\Red-DiscordBot\Red-DiscordBot\data\bestbot\cogs\CogManager\cogs\lovecalculator\lovecalculator.py", line 30, in lovecalculator
    soup_object = BeautifulSoup(await response.text(), "html.parser")
  File "c:\programs\redenv\lib\site-packages\aiohttp\client_reqrep.py", line 1009, in text
    await self.read()
  File "c:\programs\redenv\lib\site-packages\aiohttp\client_reqrep.py", line 973, in read
    self._body = await self.content.read()
  File "c:\programs\redenv\lib\site-packages\aiohttp\streams.py", line 358, in read
    block = await self.readany()
  File "c:\programs\redenv\lib\site-packages\aiohttp\streams.py", line 380, in readany
    await self._wait('readany')
  File "c:\programs\redenv\lib\site-packages\aiohttp\streams.py", line 296, in _wait
    await waiter
aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed`

--- a/lovecalculator/lovecalculator.py
+++ b/lovecalculator/lovecalculator.py
@@ -36,14 +36,16 @@ class LoveCalculator(Cog):
             async with session.get(url) as response:
                 assert response.status == 200
                 resp = await response.text()
+
         log.debug(f"{resp=}")
         soup_object = BeautifulSoup(resp, "html.parser")
-        try:
-            description = (
-                soup_object.find("div", class_="result__score").get_text().strip()
-            )
-        except:
+
+        description = soup_object.find("div", class_="result__score").get_text()
+
+        if description is None:
             description = "Dr. Love is busy right now"
+        else:
+            description = description.strip()
 
         result_image = soup_object.find("img", class_="result__image").get("src")
 
@@ -61,6 +63,11 @@ class LoveCalculator(Cog):
         except:
             title = "Dr. Love has left a note for you."
 
-        em = discord.Embed(title=title, description=result_text, color=discord.Color.red(), url=f"https://www.lovecalculator.com/{result_image}")
+        em = discord.Embed(
+            title=title,
+            description=result_text,
+            color=discord.Color.red(),
+            url=f"https://www.lovecalculator.com/{result_image}",
+        )
 
         await ctx.send(embed=em)

--- a/lovecalculator/lovecalculator.py
+++ b/lovecalculator/lovecalculator.py
@@ -1,8 +1,12 @@
+import logging
+
 import aiohttp
 import discord
 from bs4 import BeautifulSoup
 from redbot.core import commands
 from redbot.core.commands import Cog
+
+log = logging.getLogger("red.fox_v3.chatter")
 
 
 class LoveCalculator(Cog):
@@ -28,17 +32,22 @@ class LoveCalculator(Cog):
         url = "https://www.lovecalculator.com/love.php?name1={}&name2={}".format(
             x.replace(" ", "+"), y.replace(" ", "+")
         )
-        async with aiohttp.ClientSession() as session:
+        async with aiohttp.ClientSession(headers={"Connection": "keep-alive"}) as session:
             async with session.get(url) as response:
+                log.debug(f"{response=}")
+                assert response.status == 200
                 soup_object = BeautifulSoup(await response.text(), "html.parser")
                 try:
                     description = (
-                        soup_object.find("div", attrs={"class": "result__score"})
-                        .get_text()
-                        .strip()
+                        soup_object.find("div", class_="result__score").get_text().strip()
                     )
                 except:
                     description = "Dr. Love is busy right now"
+
+                result_image = soup_object.find("img", class_="result__image").get("src")
+
+                result_text = soup_object.find("div", class_="result-text").get_text()
+                result_text = " ".join(result_text.split())
 
         try:
             z = description[:2]
@@ -47,11 +56,12 @@ class LoveCalculator(Cog):
                 emoji = "❤"
             else:
                 emoji = "💔"
-            title = "Dr. Love says that the love percentage for {} and {} is:".format(x, y)
+            title = f"Dr. Love says that the love percentage for {x} and {y} is: {emoji} {description} {emoji}"
         except:
-            emoji = ""
             title = "Dr. Love has left a note for you."
 
-        description = emoji + " " + description + " " + emoji
-        em = discord.Embed(title=title, description=description, color=discord.Color.red())
+        em = discord.Embed(title=title, description=result_text, color=discord.Color.red())
+        if result_image:
+            em.set_image(url=f"https://www.lovecalculator.com/{result_image}")
+
         await ctx.send(embed=em)

